config/routes.rb

```diff
@@ -83,6 +83,8 @@
     put :unlock_access, on: :member
     post :update_locale, on: :collection
     post :resend_confirmation, on: :member
+    delete :archive, on: :member
```

Comment:
HTTP verbs delete means that there is record will be removed from the server, in our case archive mean update the attribute so better used the HTTP verbs PUT instead.